.github/workflows/check-tidy.yml

```diff
@@ -0,0 +1,59 @@
+# Copyright 2024 Democratized Data Foundation
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.txt.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0, included in the file
+# licenses/APL.txt.
+
+# This workflow checks that go mod tidy command we have set for the specific
+# go version is not broken, for example `go mod tidy -go=1.21.3`. This
+# can cause some head scratching at times, so better catch this in the PR.
+#
+# Inaddition to that also checks that we are currently in a `tidy` state.
+name: Check Tidy Workflow
+
+on:
+  pull_request:
+    branches:
+      - master
+      - develop
+
+  push:
+    tags:
+      - 'v[0-9]+.[0-9]+.[0-9]+'
+    branches:
+      - master
+      - develop
+
+jobs:
+  check-tidy:
+    name: Check mod tidy job
```

Comment:
thought: I thought you just condensed all the `check changes` into a single workflow?  Would this not be better within that file?

Reply:
It wasn't all checks that I moved in that file. Only the documentation related checks were moved there.

Reply:
Ah okay nice, thanks for the clarification/reminder :)